__tests__/invoices/invoice-details.spec.ts

```diff
@@ -0,0 +1,193 @@
+          invoiceType: InvoiceType.Lightning,
+          unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+          settlementAmount: { amount: 1, currency: WalletCurrency.Usd },
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            expiration: expect.any(Date),
+            invoiceDisplay: usdAmountInvoice,
+          }),
+        }),
+      )
+    })
+  })
+
+  describe("with btc receiving wallet", () => {
+    const btcWalletParams = {
+      ...defaultParams,
+      receivingWalletDescriptor: { id: "wallet-id", currency: WalletCurrency.Btc },
+    }
+
+    it("creates no amount lightning invoice", async () => {
+      const btcNoAmountLightningParams: CreateInvoiceDetailsParams<
+        typeof WalletCurrency.Btc
+      > = {
+        ...btcWalletParams,
+        invoiceType: InvoiceType.Lightning,
+      }
+
+      const invoiceDetails = createInvoiceDetails(btcNoAmountLightningParams)
+
+      expect(invoiceDetails).toEqual(
+        expect.objectContaining({
+          ...defaultExpectedInvoiceDetails,
+          invoiceType: InvoiceType.Lightning,
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            invoiceDisplay: noAmountInvoice,
+          }),
+        }),
+      )
+    })
+
+    it("create amount lightning invoice", async () => {
+      const btcAmountLightningParams: CreateInvoiceDetailsParams<
+        typeof WalletCurrency.Btc
+      > = {
+        ...btcWalletParams,
+        invoiceType: InvoiceType.Lightning,
+        unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+      }
+
+      const invoiceDetails = createInvoiceDetails(btcAmountLightningParams)
+
+      expect(invoiceDetails).toEqual(
+        expect.objectContaining({
+          ...defaultExpectedInvoiceDetails,
+          invoiceType: InvoiceType.Lightning,
+          unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+          settlementAmount: { amount: 1, currency: WalletCurrency.Btc },
+        }),
+      )
+
+      const invoice = await invoiceDetails.generateInvoice(mockMutations)
+
+      expect(invoice).toEqual(
+        expect.objectContaining({
+          invoice: expect.objectContaining({
+            invoiceDisplay: btcAmountInvoice,
+          }),
+        }),
+      )
+    })
+
+    it("creates an onchain address", async () => {
+      const btcOnchainParams: CreateInvoiceDetailsParams<typeof WalletCurrency.Btc> = {
+        ...btcWalletParams,
+        invoiceType: InvoiceType.OnChain,
+        unitOfAccountAmount: { amount: 1, currency: WalletCurrency.Usd },
+      }
+
+      const invoiceDetails = createInvoiceDetails(btcOnchainParams)
```

Comment:
is `createInvoiceDetails` a good fn name if it's also use for onchain?

Reply:
Yeah I struggled finding the terminology for it, lets discuss it in the mobile sync. I thought potentially it made sense because invoice could be a catch all for either an address or a payment request.